osf/models/provider.py

```diff
@@ -193,3 +199,24 @@ def create_provider_notification_subscriptions(sender, instance, created, **kwar
             event_name='new_pending_submissions',
             provider=instance
         )
+
+@receiver(post_save, sender=CollectionProvider)
+def create_primary_collection_for_provider(sender, instance, created, **kwargs):
+    if created:
+        Collection = apps.get_model('osf.Collection')
+        OSFUser = apps.get_model('osf.OSFUser')
+        _, user_id = get_request_and_user_id()
+        user = OSFUser.load(user_id)
+        if user:
+            c, created = Collection.objects.get_or_create(
+                title='{}\'s Primary Collection'.format(instance.name),
+                creator=user,
+                provider=instance,
+                is_promoted=True,
+                is_public=True
+            )
+            instance.primary_collection = c
+            instance.save()
+        else:
+            # A user is required for Collections / Groups
```

Comment:
Should this ever happen under normal circumstances? If not, then it's a bad idea to pass silently here.

Reply:
I'm... not sure. 

Currently, `PreprintProvider`'s are created through the admin interface. Since we no longer use different types of users ( 🙄) for that authentication, this shouldn't be hit, because auth is required. I expect we'll never expose API endpoints to create `*Provider`s, but it's possible that scripts (e.g. `populate_preprint_providers`) would cause this. Also, tests definitely do.